knimin/lib/data_access.py

```diff
@@ -2404,6 +2405,58 @@ def set_deposited_ebi(self):
                  WHERE barcode IN %s"""
         self._con.execute(sql, [barcodes])
 
+    def _study_exists(self, study_id):
+        """Confirms that a study ID exists
+
+        Parameters
+        ----------
+        study_id : int
+
+        Raises
+        ------
+        ValueError
+            If the study ID does not exist
+        """
+        sql = """SELECT EXISTS (SELECT 1 FROM pm.study WHERE study_id = %s)"""
+        if not self._con.execute_fetchone(sql, [study_id])[0]:
+            raise ValueError('Study ID %s does not exist.' % study_id)
+
+    def _study_is_unique(self, qiita_study_id=None, title=None, study_id=None):
+        """Confirms that a Qiita study ID and/or a title is not duplicate
+
+        Parameters
+        ----------
+        qiita_study_id : int
+        title : str
+            Search for these values to make sure they don't already exist
+        study_id : int
+            Skip this ID in searching
+
+        Raises
+        ------
+        ValueError
+            If neither value is given, or either value already exists
+        """
+        cols = ((qiita_study_id, 'qiita_study_id', 'Qiita study ID'),
```

Comment:
minor and non-blocking, but a different take on this:

```python
if qiita_study_id is None and title is None:
    # ... raise value error

for id_, column in zip((qiita_study_id, 'qiita_study_id'), (title, 'title')):
    # do stuff
```

...I don't think the description of the column adds much in my opinion